src/assets/wise5/common/save-time-message/save-time-message.component.ts

```diff
@@ -17,18 +17,56 @@ export class SaveTimeMessageComponent {
   message: string;
   tooltip: string;
 
-  constructor(private UtilService: UtilService) {}
+  constructor(@Inject(LOCALE_ID) private localeID: string) {}
 
   ngOnChanges() {
-    this.tooltip = this.UtilService.getSaveTimeText(this.saveTime, true);
+    this.tooltip = this.getSaveTimeText(this.saveTime, true);
     if (this.timeOnly) {
-      this.message = this.UtilService.getSaveTimeText(this.saveTime, this.isInactive);
+      this.message = this.getSaveTimeText(this.saveTime, this.isInactive);
     } else if (this.isSubmit) {
-      this.message = this.UtilService.getSubmittedMessage(this.saveTime, this.isInactive);
+      this.message = this.getSubmittedMessage(this.saveTime, this.isInactive);
     } else if (this.isAutoSave) {
-      this.message = this.UtilService.getSubmittedMessage(this.saveTime, this.isInactive);
+      this.message = this.getAutoSavedMessage(this.saveTime, this.isInactive);
+    } else {
+      this.message = this.getSavedMessage(this.saveTime, this.isInactive);
+    }
+  }
+
+  private getSavedMessage(clientSaveTime: number, showFullDate: boolean): string {
+    return this.getMessageText('Saved', clientSaveTime, showFullDate);
+  }
+
+  private getAutoSavedMessage(clientSaveTime: number, showFullDate: boolean): string {
+    return this.getMessageText('Auto Saved', clientSaveTime, showFullDate);
+  }
+
+  private getSubmittedMessage(clientSaveTime: number, showFullDate: boolean): string {
+    return this.getMessageText('Submitted', clientSaveTime, showFullDate);
+  }
+
+  private getMessageText(prefix: string, clientSaveTime: number, showFullDate: boolean): string {
+    const saveTimeText = this.getSaveTimeText(clientSaveTime, showFullDate);
+    return $localize`${prefix} ${saveTimeText}:saveTime:`;
+  }
```

Comment:
Do we need these intermediate functions? I think setting ```this.message``` the variables directly in ngOnChanges() would be and easier to read because they can all be written in one line.

I'm thinking something like this (this has not been tested):
```
ngOnChanges() {
    this.tooltip = this.getSaveTimeText(this.saveTime, true);
    const saveTimeText = this.getSaveTimeText(this.saveTime, this.isInactive);
    if (this.timeOnly) {
      this.message = saveTimeText;
    } else if (this.isSubmit) {
      this.message = $localize`Submitted ${saveTimeText}:saveTime:`;
    } else if (this.isAutoSave) {
      this.message = $localize`Auto Saved ${saveTimeText}:saveTime:`;
    } else {
      this.message = $localize`Saved ${saveTimeText}:saveTime:`;
    }
}
```